sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/SpendDialogPresenter.java

```diff
@@ -83,8 +84,12 @@ public void onResponse(OpenOrder response) {
 			@Override
 			public void onFailure(KinEcosystemException exception) {
 				showToast("Oops something went wrong...");
-				eventLogger
-					.send(SpendOrderCreationFailed.create(exception.getCause().getMessage(), offer.getId(), false));
+				try {
+					String errorMsg = ((ApiException) exception.getCause()).getResponseBody().getMessage();
+					eventLogger.send(SpendOrderCreationFailed.create(errorMsg, offer.getId(), false));
+				} catch (ClassCastException e) {
+					eventLogger.send(SpendOrderCreationFailed.create(exception.getMessage(), offer.getId(), false));
```

Comment:
it will be more readable (IMO) if try catch will only affect String errorMsg and after the try catch you will call 'eventLogger.send(SpendOrderCreationFailed.create(errorMsg, offer.getId(), false));'  once.

Reply:
👍